orc8r/cloud/go/services/bootstrapper/servicers/registration/cloud_registration.go

```diff
@@ -139,7 +147,16 @@ func (c *cloudRegistrationServicer) generateAndSaveTokenInfo(networkID string, l
 	return tokenInfo, nil
 }
 
-// TODO(#10437)
-func getDomainName() string {
-	return NotImplementedWarning
+func getDomainName() (string, error) {
+	// Set env variable manually if running under test conditions
+	if strings.HasSuffix(os.Args[0], ".test") {
+		os.Setenv(Orc8rDomainNameKey, Orc8rDomainNameTest)
+		return Orc8rDomainNameTest, nil
+
+	}
```

Comment:
This is so sketch, but I tried setting the Env variable in the unit tests, but it doesn't seem to get passed along to here. I'm not sure what environment the tests are running in

Reply:
If you can't test this function properly, it's probably better to just mock out the entirety of `getDomainName` during tests, instead of including this janky logic here

Reply:
That said, it's a good investment to figure out the peculiarities of the test environment (other runtime env, etc.), otherwise we risk having workarounds like these pile up, or worse, making poor architectural decisions.

Reply:
Ha, yes, this is pretty jank

Instead, let's pass in the domain name to the servicer upon servicer creation. Then have 1 constructor for the servicer that takes the domain name as a string arg, and 1 "helper" constructor that (a) reads the domain name from the env var then (b) calls the normal constructor by passing in the read value

Reply:
Ok cool, that sounds much better, thanks for suggestion 😊